infrastructure/backups/backup.sh

```diff
@@ -263,37 +263,17 @@ create_elasticsearch_backup() {
-#---------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-if  [ "$IS_LOCAL" = true ]; then
-  INFLUXDB_CONTAINER_ID=$(docker ps -aqf "name=influxdb")
-else
-  INFLUXDB_CONTAINER_ID=$(echo $(docker service ps --no-trunc -f "desired-state=running" opencrvs_influxdb) | awk '{print $11}')
-  INFLUXDB_CONTAINER_NAME=$(echo $(docker service ps --no-trunc -f "desired-state=running" opencrvs_influxdb) | awk '{print $12}')
-  INFLUXDB_HOSTNAME=$(echo $(docker service ps -f "desired-state=running" opencrvs_influxdb) | awk '{print $14}')
-  INFLUXDB_HOST=$(docker node inspect --format '{{.Status.Addr}}' "$HOSTNAME")
-  INFLUXDB_SSH_USER=${INFLUXDB_SSH_USER:-root}
-  OWN_IP=$(hostname -I | cut -d' ' -f1)
-fi
-
 # If required, SSH into the node running the opencrvs_metrics container and backup the metrics data into an influxdb subfolder
 #-----------------------------------------------------------------------------------------------------------------------------
 
 if [ "$IS_LOCAL" = true ]; then
+  INFLUXDB_CONTAINER_ID=$(docker ps -aqf "name=influxdb")
   echo "Backing up Influx locally $INFLUXDB_CONTAINER_ID"
   docker exec $INFLUXDB_CONTAINER_ID influxd backup -portable -database ocrvs /home/user/${LABEL:-$BACKUP_DATE}
   docker cp $INFLUXDB_CONTAINER_ID:/home/user/${LABEL:-$BACKUP_DATE} $ROOT_PATH/backups/influxdb/${LABEL:-$BACKUP_DATE}
-elif [[ "$OWN_IP" = "$INFLUXDB_HOST" ]]; then
-  echo "Backing up Influx on own node"
-  docker exec $INFLUXDB_CONTAINER_NAME.$INFLUXDB_CONTAINER_ID influxd backup -portable -database ocrvs /home/user/${LABEL:-$BACKUP_DATE}
-  docker cp $INFLUXDB_CONTAINER_NAME.$INFLUXDB_CONTAINER_ID:/home/user/${LABEL:-$BACKUP_DATE} $ROOT_PATH/backups/influxdb/${LABEL:-$BACKUP_DATE}
 else
-  echo "Backing up Influx on other node $INFLUXDB_HOST"
-  rsync -a -r --ignore-existing --progress --rsh="ssh -p$SSH_PORT" $ROOT_PATH/backups/influxdb $INFLUXDB_SSH_USER@$INFLUXDB_HOST:/data/backups/influxdb
-  ssh $INFLUXDB_SSH_USER@$INFLUXDB_HOST "docker exec $INFLUXDB_CONTAINER_NAME.$INFLUXDB_CONTAINER_ID influxd backup -portable -database ocrvs /home/user/${LABEL:-$BACKUP_DATE}"
-  ssh $INFLUXDB_SSH_USER@$INFLUXDB_HOST "docker cp $INFLUXDB_CONTAINER_NAME.$INFLUXDB_CONTAINER_ID:/home/user/${LABEL:-$BACKUP_DATE} /data/backups/influxdb/${LABEL:-$BACKUP_DATE}"
-  echo "Replacing backup for influxdb on manager node with new backup"
-  rsync -a -r --ignore-existing --progress --rsh="ssh -p$SSH_PORT" $INFLUXDB_SSH_USER@$INFLUXDB_HOST:/data/backups/influxdb $ROOT_PATH/backups/influxdb
+  echo "Backing up Influx in remote environment"
+  docker run --rm -v $ROOT_PATH/backups/influxdb/${LABEL:-$BACKUP_DATE}:/backup --network=$NETWORK influxdb:1.8.0 influxd backup -portable -host influxdb:8086 /backup
```

Comment:
@rikukissa does this new docker image not also need access to this volume? '/data/influxdb:/var/lib/influxdb' in order to be able to access the live data to the backups folder?  Does it need the equivalent of these props too:

```
placement:
        constraints:
          - node.labels.data1 == true
configs:
      - source: influxdb-conf.{{ts}}
        target: /etc/influxdb/influxdb.conf
```

Same required below.  I read that the --rm flag deletes the container when it exits, but wouldnt this container stay running until explicitly exited by some process?  Just wondering if you have tested this and if the container remains running afterwards. We could end up with lots of influxdb containers running as this script runs every night.

Reply:
@euanmillar this command spins up a new influxdb container that remotely connects to the actual influxdb container. For this reason it doesn't need access to the /data/influxdb directly. 

The --rm flag indeed instructs Docker to automatically remove the container when it exits. However, the behavior of the container (whether it stays running or exits immediately after execution) depends on the command it's running. In this case, the command being executed inside the container is a one-off command that exits right away after succeeding / failing. So it's in that sense identical to the `docker run mongo:4.4` and `docker run appropriate/curl` commands in this script.  If it was a container that we'd want to stay running in the background a detach flag `-d` or `--detach` would have to be supplied. 

I did run this command on farajaland-dev a few times when I was testing it
```
riku@farajaland-dev:~$ docker ps | grep influxdb
73ec6bb5d991   influxdb:1.8.0                                          "/entrypoint.sh infl…"   43 hours ago   Up 43 hours   8086/tcp                                                                   opencrvs_influxdb.1.kuzt716nr1blcpg4aceqn7nlj
```

Reply:
Amazing, thanks very much!